File: employee/employee/src/main/java/com/employee/employee/service/EmployeeService.java
package com.employee.employee.service;

import java.util.List;

import com.employee.employee.models.Employee;

public interface EmployeeService {

	Employee create(Employee emp);
	
	Employee getEmployeeById(Long id);
	
	List<Employee> getAllEmployee();
	
	void deleteEmployee(Long id);
	
	Employee update(Employee emp);
	
	
}

File: project/src/main/kotlin/com/project/service/EmployeeService.kt
package com.project.service

import com.project.domain.Employee
import com.project.repository.EmployeRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class EmployeeService(@Autowired var repositry: EmployeRepository) {

    // get all
    fun getAllRecords(): List<Employee> {
        return repositry.findAll()
    }

    //get by id
    fun getRecordById(id: Long): Employee {
        return repositry.findById(id).get()
    }

    // create record

    fun createRecord(emp: Employee): Employee {
        return repositry.save(emp)
    }

    // delete by id
    fun deleteRecord(id: Long) {
        repositry.deleteById(id)
    }

    // update
    fun updateRecord(emp: Employee): Employee {
        val emp1 = repositry.findById(emp.id!!).get()
        emp1.name = emp.name
        return repositry.save(emp1)
    }
}
